### ISSUE TITLE
execution_timeout instead of timeout with `ExternalTaskSensor` when sensing task_id

### ISSUE THREAD

**Issue by park-peter** [astronomer/astronomer-providers#857]
When using `ExternalTaskSensor` and providing `external_task_id`, it uses `execution_timeout` as the timeout and `timeout` gets ignored. The non-async sensor uses timeout for its operation so there is a disconnect with its behavior.
https://github.com/astronomer/astronomer-providers/blob/fc5b37accf37103ec6b2d9eea24e5acde1283441/astronomer/providers/core/sensors/external_task.py#L35-L53

Shouldn't the line 53 be:
```
                timeout=datetime.timedelta(seconds=self.timeout) if self.timeout != conf.getfloat("sensors", "default_timeout") else self.execution_timeout,
```
or just use `timeout`? `execution_timeout` is None by default and the sensor will run infinitely. Maybe there is a reason why this is implemented this way, but it's not explained clearly. There is a customer who has raised a Zendesk ticket (https://astronomer.zendesk.com/agent/tickets/15817) about this issue.

**Comment by rajaths010494** [comment]
<img width="806" alt="image" src="https://user-images.githubusercontent.com/92459020/214499326-d1b2e839-0e9f-4429-b584-bcf6466d34da.png">
by default, all sensors have a timeout from the BaseSensorOperator(sensors are inherited from this) which has been used in providers.

**Comment by park-peter** [comment]
@rajaths010494 correct. But it's using `execution_timeout` instead of `timeout` only when `external_task_id` is set which is inconsistent.

**Comment by rajaths010494** [comment]
Ok got it. Will change that to `timeout` instead of `execution_timeout`.

**Comment by rajaths010494** [comment]
Have raised a [PR](https://github.com/astronomer/astronomer-providers/pull/858) for using `timeout` instead of `execution_timeout` in `ExternalTaskSensor`.

**Comment by park-peter** [comment]
@rajaths010494 @phanikumv Can you please include this fix in 1.15.0? Thank you.
